### PR TITLE
Add stylelint config

### PR DESCRIPTION
**Pull request by chris-lawton**
- Adds rules from https://github.com/wagtail/wagtail/blob/main/.stylelintrc.js
- Ignore vendor dir
- Lint CSS

### CODE REVIEW

File: .stylelintrc.js [hunk 1]
```diff
@@ -1,3 +1,64 @@
 module.exports = {
   extends: '@wagtail/stylelint-config-wagtail',
+  rules: {
+    'scss/at-rule-no-unknown': [
+      true,
+      {
+        ignoreAtRules: [
+          'tailwind',
```
Comment (chris-lawton) [inline .stylelintrc.js:8]: Wonder if we should remove the tailwind references here as we (currently) don't use it.

Comment (William-Blackie) [reply .stylelintrc.js:8]: Yeah, we don't use it so no reason to include it.

File: .stylelintrc.js [hunk 2]
```diff
@@ -1,3 +1,64 @@
...
+      },
+    ],
+    'no-invalid-position-at-import-rule': [
+      true,
+      {
+        ignoreAtRules: ['tailwind', 'use'],
+      },
+    ],
+    // Would be valuable for strict BEM components but is too hard to enforce with legacy code.
+    'no-descending-specificity': null,
```
Comment (William-Blackie) [inline .stylelintrc.js:24]: Note: As we're retooling, we could enforce it now?

**Review by William-Blackie — COMMENTED** [review]
Hey, just a couple of tweaks I think @chris-lawton. Nice work.

File: .stylelintrc.js (hunk 2 above)
Comment (chris-lawton) [reply .stylelintrc.js]: Sounds good, i'll remove it

**Pull request merged**